Bottom: 22d994e31f651de6ba8e7d4510172361f48f43ef
Top:    5e91aa2ed0c3c4e59431c29d53601179f431dac3
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2009-01-28 22:02:02 +0000

Workaround crash where contour is passed as NULL to poly_PreContour









---

diff --git a/src/polygon.c b/src/polygon.c
index 43432bc..7410329 100644
--- a/src/polygon.c
+++ b/src/polygon.c
@@ -247,6 +247,16 @@ original_poly (PolygonType * p)
       }
   }
   END_LOOP;
+  if (contour == NULL)
+    {
+      printf ("How did that escape - did the loop iterate zero times??\n");
+      POLYGONPOINT_LOOP (p);
+        {
+          printf ("Hello\n");
+        }
+      END_LOOP;
+      return NULL;
+    }
   poly_PreContour (contour, TRUE);
   /* make sure it is a positive contour */
   if ((contour->Flags.orient) != PLF_DIR)
